src/v/cloud_storage/tests/segment_meta_cstore_fuzz.cc

```diff
@@ -424,27 +424,60 @@ using cstore_operation = std::variant<
 // SIN SECTION
 // this is a sections of ODR sins to appease the daemons of the linker
 
-auto cloud_storage::operator<<(std::ostream& os, segment_meta const& s)
-  -> std::ostream& {
-    return os << fmt::format(
-             "{{is_compacted: {}, size_bytes: {}, base_offset: {}, "
-             "committed_offset: "
-             "{}, base_timestamp: {}, max_timestamp: {}, delta_offset: {}, "
-             "ntp_revision: {}, archiver_term: {}, segment_term: {}, "
-             "delta_offset_end: {}, sname_format: {}, metadata_size_hint: {}}}",
-             s.is_compacted,
-             s.size_bytes,
-             s.base_offset,
-             s.committed_offset,
-             s.base_timestamp,
-             s.max_timestamp,
-             s.delta_offset,
-             s.ntp_revision,
-             s.archiver_term,
-             s.segment_term,
-             s.delta_offset_end,
-             s.sname_format,
-             s.metadata_size_hint);
+auto fmt::formatter<cloud_storage::segment_meta>::format(
+  cloud_storage::segment_meta const& s, format_context& ctx) const
```

Comment:
looks like you could move `segment_meta` out of the generic types.h and it wouldn't have much of any redpanda dependencies. then maybe you wouldn't need to duplicate this.

Reply:
i don't see any reason why it can't live in its own header.

Reply:
done